### PR TITLE
add thirdreality multi-function light

### PR DESCRIPTION
**Pull request by weihuan1111**
Change-Id: I188792e689ada8ee624eb3ba11492419852a2303

### CODE REVIEW

**Comment by weihuan1111** [comment]
@brianj94 Hi, we have submit the certification for our three devices, but now, we have not receive the e-mail from UL to do WWST certification, is there any problem with the certification we submitted?
7e01fefbf4f6b392856f7445a1a7953

**Comment by Brianj94** [comment]
Hi @weihuan1111 , 

Thank you for following up on these device certifications. Currently we are still working through our internal code review process for these submissions. Once those are completed we will create the necessary tickets on our side to bring in UL and have their team scope testing, quotes, and a timeline.  

At this time there are no additional steps needed from your team. 

Thank you,
Brian J

File: drivers/SmartThings/zigbee-switch/src/thirdreality/init.lua [hunk 1]
```diff
@@ -0,0 +1,31 @@
+local capabilities = require "st.capabilities"
+
+local THIRDREALITY_MOTIOM_CLUSTER = 0xFC00
```
Comment (ctowns) [inline drivers/SmartThings/zigbee-switch/src/thirdreality/init.lua:3]: Fix spelling here and below
```suggestion
local THIRDREALITY_MOTION_CLUSTER = 0xFC00
```

File: drivers/SmartThings/zigbee-switch/src/thirdreality/init.lua [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+local ATTR_TR = 0x0002
+local MOTION_DETECT = 0x0001
+local MOTION_NO_DETECT = 0x0000
+
+local function present_value_attr_handler(driver, device, value, zb_rx)
```
Comment (ctowns) [inline drivers/SmartThings/zigbee-switch/src/thirdreality/init.lua:8]: rename this function to something more accurate like `motion_sensor_attr_handler` or `occupancy_attr_handler`

File: drivers/SmartThings/zigbee-switch/src/thirdreality/init.lua [hunk 3]
```diff
@@ -0,0 +1,31 @@
+local capabilities = require "st.capabilities"
+
+local THIRDREALITY_MOTIOM_CLUSTER = 0xFC00
+local ATTR_TR = 0x0002
```
Comment (ctowns) [inline drivers/SmartThings/zigbee-switch/src/thirdreality/init.lua:4]: rename this constant something more descriptive like `MOTION_SENSOR_VALUE`

**Review by ctowns — COMMENTED** [review]
A couple small changes. Also, please add some unit tests so that this code has proper coverage

**Comment by weihuan1111** [comment]
> A couple small changes. Also, please add some unit tests so that this code has proper coverage

@ctowns How to add unit tests? And how to test the unit tests?

**Comment by lelandblue** [comment]
Hello @weihuan1111 . 

Yes, unit test coverage is necessary. Please see the other comments on the 2 other Third Reality requests for more details on this topic. 


Thank you

**Comment by weihuan1111** [comment]
@lelandblue Test units has been added.

**Comment by weihuan1111** [comment]
@greens Test units has been added.

**Comment by greens** [comment]
@weihuan1111 your tests do not run for the same reason as in https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/772

**Comment by weihuan1111** [comment]
@greens Fixed, please review it, thank you.

**Review by greens — CHANGES_REQUESTED** [review]
Please run your own tests locally before committing them here. We have provided you guides on how to do this.

Your tests do not run. I will not be telling you why because you should be running them yourself.

**Comment by weihuan1111** [comment]
@greens @Brianj94

**Comment by greens** [comment]
@weihuan1111 If you are running locally and things are working it's because you have a file locally that is not a part of your pull request. Your tests, using only the files in your PR, still do not run.

The profile `on-off-level-rgbw-motion-sensor` referenced in your tests does not exist.

**Comment by weihuan1111** [comment]
@greens Fixed.

**Comment by greens** [comment]
Your tests still do not run. Can you explain to me the issues you're having setting up the tests to run locally? If you tell me the problems you're having, maybe I can help.

**Comment by weihuan1111** [comment]
@greens  I found the problem, i lose the .yml in get_profile_definition.

**Comment by weihuan1111** [comment]
@greens I still have question. why i run other test, like test_on_off_zigbee_bulb.lua , still show "\profiles\on-off-level-intensity.yml: No such file or directory"

6b48bfa740adeb9e4ebff45297a7fe7

**Comment by greens** [comment]
@weihuan1111 tests should be run from the `/src` directory of the driver

File: drivers/SmartThings/zigbee-switch/src/thirdreality/init.lua [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+local capabilities = require "st.capabilities"
+
+local THIRDREALITY_MOTION_CLUSTER = 0xFC00
```
Comment (greens) [inline drivers/SmartThings/zigbee-switch/src/thirdreality/init.lua:17]: you don't need to bind to or configure reporting for this cluster?

**Review by greens — APPROVED** [review]

**Comment by weihuan1111** [comment]
@Brianj94 Thank you, i have add the Firmware version and Product Category for the device in this pull request.

**Comment by Brianj94** [comment]
Hi @weihuan1111, I hope you are doing well. I wanted to follow up on this device certification as well as the Vibration Sensor. I see that it has not moved into testing with UL and I wanted to see if you've received the quote from UL and if samples have been sent. Please let me know if you have any info on the current status of this device. 

If samples have been sent and quote have been completed please let me know, and I will follow up with the UL team. 

Thank you,
Brian J

**Comment by weihuan1111** [comment]
@Brianj94 The button and vibration sensor have been sent to UL, and UL have received the devices.
We have received  the quote of the night light from UL, we have not sent the sample.

**Comment by Brianj94** [comment]
@weihuan1111 Okay, thank you for the update. I will push from my side to have the sensor and button move along through testing. Hope you have a good weekend.

**Comment by Brianj94** [comment]
Hi @weihuan1111 , I hope you are well. Can you please provide a timeline for when the Night Light quote will be processed and samples provided to UL?

**Comment by lelandblue** [comment]
The status of this request has changed. If you have questions please reach out to partners@smartthings.com

Thank you.

**Pull request closed without merging**